### PR TITLE
added component graph and callback function for sidebar

### PR DESCRIPTION
**Pull request by micahcarroll**
#90

### CODE REVIEW

File: app/assets/javascripts/course/components/ComponentGraph.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    super(props)
+
+    this.state = { subsection: {} }
+
+    this.requestSubsection(this.props.id)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.requestSubsection(nextProps.id)
+    console.log(nextProps);
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/ComponentGraph.js]: remove log

File: app/assets/javascripts/course/components/CoursePage.js [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 import CourseSidebar from './CourseSidebar'
+import ComponentGraph from './ComponentGraph'
 
 class CoursePage extends React.Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = { subsection_displayed: 31 }
+
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/CoursePage.js]: space

File: app/assets/javascripts/course/components/CoursePage.js [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
+  changeSubsectionDisplayed(id) {
+    this.setState({ subsection_displayed: id})
+  }
+
   render() {
     return (
       <div>
         <h1>This is a course page</h1>
-        <CourseSidebar />
+        <CourseSidebar id={2} callback={this.changeSubsectionDisplayed.bind(this)}/>
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/CoursePage.js]: remove placeholder

**Comment by micahcarroll** [comment]
made changes

**Comment by charlesx2013** [comment]
👍

**Pull request merged**